tsdb/shard.go

```diff
@@ -284,6 +293,11 @@ func (s *Shard) WritePoints(points []models.Point) error {
 
 // DeleteSeries deletes a list of series.
 func (s *Shard) DeleteSeries(seriesKeys []string) error {
+	s.mu.RLock()
```

Comment:
This may be the correct lock to take but L306 takes a write lock for some reason.  The code is basically doing the same thing so we should probably use the same lock for both.